core/src/main/java/org/apache/brooklyn/util/core/logbook/opensearch/OpenSearchLogStore.java

```diff
@@ -0,0 +1,228 @@
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.apache.brooklyn.util.core.logbook.LogbookConfig.BASE_NAME_LOGBOOK;
+
+/**
+ * Implementation for expose log from ElasticSearch to the logbook API.
+ */
+public class OpenSearchLogStore implements LogStore {
+
+    /*
+     # example config for local default implementation
+     brooklyn.logbook.logStore = org.apache.brooklyn.util.core.logbook.opensearch.OpenSearchLogStore
+     brooklyn.logbook.openSearchLogStore.host = https://localhost:9200
+     brooklyn.logbook.openSearchLogStore.index = brooklyn8
+     brooklyn.logbook.openSearchLogStore.user = admin
+     brooklyn.logbook.openSearchLogStore.password = admin
+     brooklyn.logbook.openSearchLogStore.verifySsl = false
+     */
+    public final static String BASE_NAME_OPEN_SEARCH_LOG_STORE = BASE_NAME_LOGBOOK + ".openSearchLogStore";
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_HOST = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".host", "Log store host");
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_INDEX = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".index", "Log store index");
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_USER = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".user", "User name");
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_PASS = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".password", "User password");
+
+    public final static ConfigKey<String> LOGBOOK_LOG_STORE_APIKEY = ConfigKeys.newStringConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".apikey", "API key");
+
+    public final static ConfigKey<Boolean> LOGBOOK_LOG_STORE_VERIFY_SSL = ConfigKeys.newBooleanConfigKey(
+            BASE_NAME_OPEN_SEARCH_LOG_STORE + ".verifySsl", "Verify SSL", true);
+
+    private final ManagementContext mgmt;
+    CloseableHttpClient httpClient;
+    private String host;
+    private String user;
+    private String password;
+    private String apiKey;
+    private Boolean verifySsl;
+    private String indexName;
+
+    @VisibleForTesting
+    public OpenSearchLogStore() {
+        this.mgmt = null;
+    }
+
+    public OpenSearchLogStore(ManagementContext mgmt) {
+        this.mgmt = mgmt;
+        initialize();
+
+        HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        if (!verifySsl) {
+
+            final SSLContext sslContext;
+            try {
+                sslContext = SSLContextBuilder
+                        .create()
+                        .loadTrustMaterial(new TrustSelfSignedStrategy())
```

Comment:
Should this be configurable?

Reply:
worth a project-wide review to make this configurable.  not appropriate IMO to figure out the project-wide solution as part of this PR.

Reply:
Yes you are right. This implementation is ok for the docker image provided by the Open Search project. For production this need to be configurable.
For now I think we are good addressing it in a future PR. What do you think @duncangrant

Reply:
I think what you have is fine